src/Groups/homomorphisms.jl

```diff
@@ -239,11 +239,13 @@ end
 
 """
     cokernel(f::GAPGroupHomomorphism)
-Return the cokernel of `f`.
+Return the cokernel of `f`, that is, the quotient of the codomain of `f`
+by the normal closure of the image.
```

Comment:
That's the definition of cokernel in category theory. It is however *not* the definition GAP uses for "cokernel" (which is the kernel of the "reverse" generalized map). 

I think it's good to have it this way, but I'd suggest we add a warning about this in the doc string, for long time GAP users. Of course that's beyond the scope of this PR, but it'd be nice nevertheless.

Reply:
Judging from the code (modulo the error), this seems to be the intended definition.
But of course we have to decide what we want to model, and then change the code if necessary.

If we do eventually want to support a machinery of multi-valued mappings like in GAP then we may take a definition as in GAP (`CoKernelOfMultiplicativeGeneralMapping`), and then just delegate to the relevant GAP functions.
If only homomorphisms are of interest then we may take the proposed definition.

Perhaps the most elegant solution is to use more specific variable names, in order to allow for several definitions.

Reply:
The category of groups has cokernels. That the cokernel object is isomorphic to the quotient of the range modulo the normal-closure of the set-theoretic image is not a definition but a proposition.

I am puzzled by the documentation of `CoKernelOfMultiplicativeGeneralMapping` maybe because I am not sure if I understand the meaning of a `MultiplicativeGeneralMapping`. What I believe to understand:
* a general mapping in `GAP` is simply a relation where the first factor of the product is called source and the second range
* a multiplicative general mapping is a relation which takes the extra structure into account, i.e., if G and H are groups then the relation must be a subgroup of G x H.

Right?

Reply:
Right. And the "Cokernel" (in the GAP sense) is simply the "mirror" of the kernel; i.e. the kernel of the generalized inverse map. Or put yet another way: the kernel is the subset of all elements (g,1) in the relation, while the cokernel is the subset  of elements (1,g) -- which for an "honest" homomorphism of course only consists of the single element (1,1)

Reply:
General mappings form a category. However, this category has neither initial nor terminal objects and hence no zero objects and no notion of a zero morphism. This also means that there is no notion of kernel or cokernel. If you wish I can think of a categorical characterization of the set-theoretic subobjects which are called Kernel and CoKernel of multiplicative general mappings in GAP.